packages/server/assets/viewer/typedefs/savedViews.graphql

```diff
@@ -0,0 +1,203 @@
+  """
+  Whether to only include views matching this search term
+  """
+  search: String
+  """
+  Optionally specify sort direction. Default: descending
+  """
+  sortDirection: SortDirection
+  """
+  Optionally specify sort by field. Default: updatedAt
+  Options: updatedAt, createdAt, name
+  """
+  sortBy: String
+  limit: Int
+  cursor: String
+}
+
+type SavedViewGroup {
+  id: ID!
+  projectId: ID!
+  modelIds: [ID!]!
+  title: String!
+  isUngroupedViewsGroup: Boolean!
+  views(input: SavedViewGroupViewsInput!): SavedViewCollection!
+}
+
+type SavedViewGroupCollection {
+  cursor: String
+  totalCount: Int!
+  items: [SavedViewGroup!]!
+}
+
+input SavedViewGroupsInput {
+  """
+  Viewer resource ID string that identifies which resources should be loaded
+  """
+  resourceIdString: String!
+  """
+  Whether to only include groups w/ views authored by the current user
+  """
+  onlyAuthored: Boolean
+  """
+  Whether to only include groups that have views matching this search term
+  """
+  search: String
+  limit: Int
+  cursor: String
+}
+
+extend type Project {
+  savedViewGroups(input: SavedViewGroupsInput!): SavedViewGroupCollection!
+}
+
+"""
+MUTATIONS:
+"""
+input CreateSavedViewInput {
+  projectId: ID!
+  resourceIdString: String!
+  """
+  Group name, if grouping necessary
+  """
+  group: String
+
+  """
+  Auto-generated name, if not specified
+  """
+  name: String
+  description: String
+  """
+  SerializedViewerState. If omitted, comment won't render (correctly) inside the
+  viewer, but will still be retrievable through the API
+  """
+  viewerState: JSONObject!
+  """
+  Encoded screenshot of the view
+  """
+  screenshot: String!
+  """
+  Optionally also set this as the home/default view for the target model
+  """
+  isHomeView: Boolean
+  """
+  Set visibility of the view. Default: public
+  """
+  visibility: SavedViewVisibility
+}
+
+input UpdateSavedViewInput {
+  id: ID!
+  """
+  New resource targets, if necessary
+  """
+  resourceIdString: String
+  """
+  New group name, if grouping necessary
+  """
+  group: String
+  """
+  New name for the view
+  """
+  name: String
+  description: String
+  """
+  SerializedViewerState. If omitted, comment won't render (correctly) inside the
+  viewer, but will still be retrievable through the API
+  """
+  viewerState: JSONObject
+  """
+  Encoded screenshot of the view
+  """
+  screenshot: String
+  """
+  Optionally also set this as the home/default view for the target model
+  """
+  isHomeView: Boolean
+  """
+  Optionally change visibility of the view
+  """
+  visibility: SavedViewVisibility
+}
+
+input DeleteViewInput {
+  id: ID!
+  projectId: ID!
+}
+
+input RenameGroupInput {
+  projectId: ID!
+  groupId: ID!
+  newName: String!
+}
+
+input DeleteViewGroupInput {
+  projectId: ID!
+  groupId: ID!
+}
+
+"""
+One or both of the position values must be specified
+"""
+input ChangeViewOrderInput {
+  id: ID!
+  projectId: ID!
+  previousViewPosition: Float
```

Comment:
gonna try something like this 

<img width="1372" height="1457" alt="Image" src="https://github.com/user-attachments/assets/6dd04d16-d70c-41b9-85b0-99597e46ed0e" />